homeassistant/components/trest_solar/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for TrestSolarController integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, Optional
+
+from trest_identity import TrestIdentityService
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class TrestSolarControllerConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TrestSolarController."""
+
+    async def async_step_user(
+        self,
+        user_input: Optional[dict[str, Any]] = None,  # noqa: UP007
```

Comment:
please use new optional notation
```suggestion
        user_input: dict[str, Any] | None = None,
```